app/models/dataload.rb

```diff
@@ -0,0 +1,16 @@
+class Dataload < ApplicationRecord
+  validates :doi, :mountPoint, :directory, :user_name, presence: true
+  validates :user_email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
+
+  enum status: { created: 0, in_progress: 1, completed: 2, failed: 3, archived: 4 }
```

Comment:
i'm not sure when this was introduced in Rails, but i think we can just use symbolized values for the enums in Rails 7.1: https://thoughtbot.com/blog/enum-validations-and-database-constraints-in-rails-7-1

Reply:
Nice and clean, I'll give that a try - thanks!